state-chain/pallets/cf-elections/src/electoral_systems/composite.rs

```diff
@@ -1,31 +1,12 @@
-use crate::electoral_system::{ElectoralSystem, ElectoralWriteAccess};
-
 /// Allows the composition of multiple ElectoralSystems while allowing the ability to configure the
 /// `on_finalize` behaviour without exposing the internal composite types.
-pub struct Composite<T, ValidatorId, H = DefaultHooks<()>> {
-	_phantom: core::marker::PhantomData<(T, ValidatorId, H)>,
-}
-
-pub struct DefaultHooks<OnFinalizeContext> {
-	_phantom: core::marker::PhantomData<OnFinalizeContext>,
-}
-
-pub trait Translator<GenericElectoralAccess> {
-	type ElectoralSystem: ElectoralSystem;
-	type ElectionAccess<'a>: ElectoralWriteAccess<ElectoralSystem = Self::ElectoralSystem>
-	where
-		Self: 'a,
-		GenericElectoralAccess: 'a;
-
-	fn translate_electoral_access<'a>(
-		&'a self,
-		generic_electoral_access: &'a mut GenericElectoralAccess,
-	) -> Self::ElectionAccess<'a>;
+pub struct CompositeRunner<T, ValidatorId, StorageAccess, H> {
+	_phantom: core::marker::PhantomData<(T, ValidatorId, StorageAccess, H)>,
 }
 
 /// The access wrappers need to impl the access traits once for each variant,
 /// these tags ensure these trait impls don't overlap.
-mod tags {
+pub mod tags {
 	pub struct A;
 	pub struct B;
```

Comment:
Sensible names is a little difficult, because every chain could be different, so if we named them "block height", "ingress" etc. depending on the chain, it could be different. Though, it's certainly on the radar as something that would be nice to improve if possible

Reply:
I guess what I mean is that we would use this macro at the runtime-level (ie. solana_elections.rs). We don't really need to define a generic that is only used once. But yeah - can wait for later ofc.